File: 129. Sum Root to Leaf Numbers.py
#Python:

class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        res = 0
        def dfs(node, num):
            nonlocal res
            if node is None:
                return
            num = num * 10 + node.val
            if node.left is None and node.right is None:
                res += num
            dfs(node.left, num)
            dfs(node.right, num)
        dfs(root, 0)
        return res


#Java:

class Solution {
    int res;
    public int sumNumbers(TreeNode root) {
        dfs(root, 0);
        return res;
    }
    private void dfs(TreeNode node, int num){
        if(node == null) return;
        num = num * 10 + node.val;
        if(node.left == null && node.right == null){
            res += num;
        }
        dfs(node.left, num);
        dfs(node.right, num);
    }
}

File: 129. Sum Root to Leaf Numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int res = 0;
    public int sumNumbers(TreeNode root) {
        helper(root, 0);
        return res;
    }
    void helper(TreeNode node, int num){
        if(node == null){
            return;
        }
        num = num * 10 + node.val;
        if(node.left == null && node.right == null) res += num;
        helper(node.left, num);
        helper(node.right, num);
    }
}
